WebRoot/js/zimbraAdmin/common/ZaTree.js

```diff
@@ -524,10 +524,21 @@ function(item, ev) {
 
         this._selectedItems.add(item);
 
-		if (item._setSelected(true)) {
-            this._updateHistory(item, true, isShowInHistory);
-			this._notifyListeners(DwtEvent.SELECTION, [item], DwtTree.ITEM_SELECTED, ev, this._selEv);
-		}
+        if (item._setSelected(true)) {
+            if (item._text === ZaMsg.NAD_Tab_HSM) {
+                if (typeof ZaHSM === 'undefined') {
+                    ZaApp.getInstance().getCurrentController().popupErrorDialog(ZaMsg.ERROR_Feature_Not_Licensed + "<br>" + ZaMsg.INFO_gotoLicenseet);
+                } else if (ZaHSM && ZaHSM.checkstoragefeaturenabled()) {
+                    this._updateHistory(item, true, isShowInHistory);
+                    this._notifyListeners(DwtEvent.SELECTION, [item], DwtTree.ITEM_SELECTED, ev, this._selEv);
+                } else {
+                    ZaApp.getInstance().getCurrentController().popupErrorDialog(ZaMsg.ERROR_Feature_Not_Licensed + "<br>" + ZaMsg.INFO_gotoLicense);
+                }
```

Comment:
I think we can remove the first `if` block now

Reply:
we can't remove it aas it will throw exception as **ReferenceError: ZaHSM is not defined at ZaTree._itemClicked** . The first if block will be executed when there is no SM admin extension is deployed onto the server.
In the existing behaviour the SM extension mainly provides support for S3 bucket and executing SM policies. So if the SM extension is not available the user is able to view/create internal volumes(loading from zm-admin-console)  but not S3 volumes.
now, as per current requirement of the ticket we are not even displaying the internal volumes to the user if the licensing attribute is false, so we need the first if condition as well.